test/tools/yulopti.cpp

```diff
@@ -162,85 +162,97 @@ class YulOpti
-			}
-			map<char, string> const& abbreviationMap = OptimiserSuite::stepAbbreviationToNameMap();
+			disambiguated = disambiguated || (disambiguate(), true);
 			map<char, string> const& extraOptions = {
 				// QUIT starts with a non-letter character on purpose to get it to show up on top of the list
 				{'#', ">>> QUIT <<<"},
 				{',', "VarNameCleaner"},
 				{';', "StackCompressor"}
 			};
 
-			printUsageBanner(abbreviationMap, extraOptions, 4);
+			printUsageBanner(extraOptions, 4);
 			cout << "? ";
 			cout.flush();
-			// TODO: handle EOF properly.
 			char option = static_cast<char>(readStandardInputChar());
 			cout << ' ' << option << endl;
 
-			OptimiserStepContext context{
-				m_dialect,
-				*m_nameDispenser,
-				reservedIdentifiers,
-				solidity::frontend::OptimiserSettings::standard().expectedExecutionsPerDeployment
-			};
-
-			auto abbreviationAndName = abbreviationMap.find(option);
-			if (abbreviationAndName != abbreviationMap.end())
+			try
 			{
-				OptimiserStep const& step = *OptimiserSuite::allSteps().at(abbreviationAndName->second);
-				step.run(context, *m_ast);
+				switch (option)
+				{
+					case 4:
+					case '#':
+						return;
+					case ',':
+						VarNameCleaner::run(m_context, *m_ast);
+						// VarNameCleaner destroys the unique names guarantee of the disambiguator.
+						disambiguated = false;
+						break;
+					case ';':
+					{
+						Object obj;
+						obj.code = m_ast;
+						StackCompressor::run(m_dialect, obj, true, 16);
+						break;
+					}
+					default:
+						OptimiserSuite{m_context}.runSequence(
+							std::string_view(&option, 1),
+							*m_ast
+						);
+				}
+				_source = AsmPrinter{m_dialect}(*m_ast);
 			}
-			else switch (option)
-			{
-			case '#':
-				return;
-			case ',':
-				VarNameCleaner::run(context, *m_ast);
-				// VarNameCleaner destroys the unique names guarantee of the disambiguator.
-				disambiguated = false;
-				break;
-			case ';':
+			catch (exception const& _exception)
 			{
-				Object obj;
-				obj.code = m_ast;
-				StackCompressor::run(m_dialect, obj, true, 16);
-				break;
+				cerr << "Exception during optimiser step: " << _exception.what() << endl;
 			}
-			default:
-				cerr << "Unknown option." << endl;
-			}
-			source = AsmPrinter{m_dialect}(*m_ast);
+			cout << "----------------------" << endl;
+			cout << _source << endl;
 		}
 	}
 
 private:
-	ErrorList m_errors;
-	shared_ptr<CharStream> m_charStream;
 	shared_ptr<yul::Block> m_ast;
 	Dialect const& m_dialect{EVMDialect::strictAssemblyForEVMObjects(EVMVersion{})};
-	shared_ptr<AsmAnalysisInfo> m_analysisInfo;
-	shared_ptr<NameDispenser> m_nameDispenser;
+	unique_ptr<AsmAnalysisInfo> m_analysisInfo;
+	set<YulString> const m_reservedIdentifiers = {};
+	NameDispenser m_nameDispenser{m_dialect, m_reservedIdentifiers};
+	OptimiserStepContext m_context{
+		m_dialect,
+		m_nameDispenser,
+		m_reservedIdentifiers,
+		solidity::frontend::OptimiserSettings::standard().expectedExecutionsPerDeployment
```

Comment:
Just a question: I just noticed that we have it hard-coded. Isn't this parameter kinda important when playing with the optimizer? Maybe we should have a prameter for this (not in this PR of course)?

Reply:
Yep, when I stumbled across this now, I also thought, it should probably be configurable - it would be a bit annoying to do in interactive mode (i.e. allow changing it between steps on the fly), but at least a command line option might be nice. But yeah, we can do it separately.